types/react-dropzone/index.d.ts

```diff
@@ -1,46 +1,41 @@
-// Type definitions for react-dropzone
+// Type definitions for react-dropzone 3.13
 // Project: https://github.com/okonet/react-dropzone
 // Definitions by: Mathieu Larouche Dube <https://github.com/matdube>, Ivo Jesus <https://github.com/LynxEyes>, Luís Rodrigues <https://github.com/goblindegook>, Ben Bayard <https://github.com/benbayard>
 // Definitions: https://github.com/Vooban/DefinitelyTyped
 // TypeScript Version: 2.3
 
-/// <reference types="react"/>
+import { CSSProperties, Component, DragEvent, InputHTMLAttributes } from "react";
 
-declare module "react-dropzone" {
-    interface DropzoneProps {
-        // Drop behavior
-        onDrop?: (accepted: File[], rejected: File[]) => any;
-        onDropAccepted?: (accepted: File[]) => any;
-        onDropRejected?: (rejected: File[]) => any;
-
-        // Drag behavior
-        onDragStart?: Function;
-        onDragEnter?: Function;
-        onDragLeave?: Function;
-
-        style?: React.CSSProperties; // CSS styles to apply
-        activeStyle?: React.CSSProperties; // CSS styles to apply when drop will be accepted
-        rejectStyle?: React.CSSProperties; // CSS styles to apply when drop will be rejected
-        className?: string; // Optional className
-        activeClassName?: string; // className for accepted state
-        rejectClassName?: string; // className for rejected state
+declare namespace Dropzone {
+    interface ImageFile extends File {
+        preview?: string;
+    }
 
-        disablePreview?: boolean; // Enable/disable preview generation
-        disableClick?: boolean; // Disallow clicking on the dropzone container to open file dialog
+    interface DropFilesEventHandler {
+        (accepted: ImageFile[], rejected?: ImageFile[]): void;
+        <E extends DragEvent<HTMLDivElement>>(event: E): void;
+    }
```

Comment:
Reading into the source, it seems this got confused between `this.onDrop` and `this.props.onDrop`. `this.onDrop` is the `DragEvent` handler, and is internal to dropzone;  the handler given on props is `(accepted: ImageFile[], rejected: ImageFile[], event: React.DragEvent<HTMLDivElement>>): void`.

It could also include the `this` type (`(this: Dropzone, ...`) but not sure it's a good idea.